integration/oidc/authorization-code/README.md

```diff
@@ -0,0 +1,197 @@
+## Overview
+This example application uses the OpenID Connect (OIDC) protocol to authenticate
+with Azure Active Directory and get an access token to request WorkflowGen's
+GraphQL API.
+
+## Prerequisites
+1. dotnet core sdk 2.2
+1. Make sure to have a valid WorkflowGen licence and serial number.
+1. Make sure to have access to an Azure subscription.
+1. Make sure to have a WorkflowGen server setup with OIDC authentication with
+Azure AD.
+
+    Read the [WorkflowGen for Azure](https://docs.advantys.com/workflowgen-for-azure/) guide to know how!
+
+## Technologies
+
+This example uses aspnet core 2.2 with the OpenIDConnect authentication
+middleware. For GraphQL requests, it uses the GraphQL.Client library. Pages are
+generated using the Razor engine. Bootstrap and jQuery are available client-side.
+It is based on the dotnet core template named `webapp`.
+
+# Setup
+Before beginning the setup of this example, you should now already have two (2)
+app registrations in your Azure AD that represents WorkflowGen's portal and
+WorkflowGen's GraphQL API. If you don't, you must finish the authentication
+section of the WorkflowGen for Azure guide.
+
+## Create a New App Registration in Azure AD (Azure cli)
+The following commands create the necessary application registration in
+Azure AD using the Azure cli commandline tools.
+
+**Note**: You need to be logged in the Azure cli. To do this, run `az login`.
+
+### Create the application registration
+```powershell
+# Replace `WorkflowGen GraphQL API` with the name of the registered WorkflowGen
+# GraphQL API application including the backticks if you named it something else.
+$webApiApp = & az ad app list `
+    --query '[?displayName == `WorkflowGen GraphQL API`] | [0]' `
+    | ConvertFrom-Json `
+    | Select-Object -First 1
+$password = [guid]::NewGuid().ToString()
+$webAppUrl = "http://localhost:8080"
+
+# Create the application registration.
+$webAppClient = & az ad app create `
+    --display-name "My aspnet core Web App" `
+    --homepage $webAppUrl `
+    --identifier-uris $webAppUrl `
+    --password $password `
+    --end-date '2299-12-31T11:59:59+00:00' `
+    --reply-urls "$webAppUrl/signin" `
+    | ConvertFrom-Json
+
+# Create a service principal for the registration
+& az ad sp create --id $webAppClient.appId
+
+# Add required permissions for the GraphQL API access
+& az ad app permission add `
+    --id $webAppClient.appId `
+    --api $webApiApp.appId `
+    --api-permissions "$($webApiApp.oauth2Permissions.id)=Scope"
+
+# Grant the required permissions for the GraphQL API Access
+& az ad app permission grant `
+    --api $webApiApp.appId `
+    --id $webAppClient.appId
+```
+
+You will need the following information for the example configuration:
+
+* The client id. This is the value of `$webAppclient.appId`.
+* The client secret. This is the value of `$password`.
+* The GraphQL API App Id URI. This is the value of `$webApiApp.identifierUris[0]`.
```

Comment:
* The GraphQL API App ID URI. This is the value of `$webApiApp.identifierUris[0]`.